internal/armada/repository/job.go

```diff
@@ -169,88 +168,82 @@ func (repo *RedisJobRepository) ReturnLease(clusterId string, jobId string) (ret
 	return nil, nil
 }
 
-func (repo *RedisJobRepository) Cancel(jobs []*api.Job) map[*api.Job]error {
+type deleteJobRedisResponse struct {
+	job                       *api.Job
+	removeFromLeasedResult    *redis.IntCmd
+	removeFromQueueResult     *redis.IntCmd
+	setJobExpiryResult        *redis.BoolCmd
+	deleteJobQueueIndexResult *redis.IntCmd
+	deleteJobSetIndexResult   *redis.IntCmd
+}
 
+func (repo *RedisJobRepository) DeleteJobs(jobs []*api.Job) map[*api.Job]error {
 	pipe := repo.db.Pipeline()
-	queueCmds := []*redis.IntCmd{}
-	for _, job := range jobs {
-		queueCmds = append(queueCmds, pipe.ZRem(jobQueuePrefix+job.Queue, job.Id))
-	}
-	leaseCmds := []*redis.IntCmd{}
+	deletionResults := make([]*deleteJobRedisResponse, 0, len(jobs))
 	for _, job := range jobs {
-		leaseCmds = append(leaseCmds, pipe.ZRem(jobLeasedPrefix+job.Queue, job.Id))
+		deletionResult := &deleteJobRedisResponse{job: job}
+		deletionResult.removeFromQueueResult = pipe.ZRem(jobQueuePrefix+job.Queue, job.Id)
+		deletionResult.removeFromLeasedResult = pipe.ZRem(jobLeasedPrefix+job.Queue, job.Id)
+		deletionResult.setJobExpiryResult = pipe.Expire(jobObjectPrefix+job.Id, time.Hour*24*7)
+		deletionResult.deleteJobQueueIndexResult = pipe.HDel(jobQueueMapKey, job.Id, job.Queue)
+		deletionResult.deleteJobSetIndexResult = pipe.SRem(jobSetPrefix+job.JobSetId, job.Id)
+		deletionResults = append(deletionResults, deletionResult)
 	}
-
 	_, _ = pipe.Exec() // ignoring error here as it will be part of individual commands
 
 	cancelledJobs := map[*api.Job]error{}
-	for i, job := range jobs {
-		result, e := queueCmds[i].Result()
-		if e != nil {
-			cancelledJobs[job] = e
-		}
-		if result > 0 {
-			cancelledJobs[job] = nil
-		}
-	}
-	for i, job := range jobs {
-		result, e := leaseCmds[i].Result()
-		if e != nil {
-			cancelledJobs[job] = e
+	for _, deletionResult := range deletionResults {
+		numberOfUpdates, err := processDeletionResponse(deletionResult)
+
+		if numberOfUpdates > 0 {
+			cancelledJobs[deletionResult.job] = nil
 		}
-		if result > 0 {
-			cancelledJobs[job] = nil
+
+		if err != nil {
+			cancelledJobs[deletionResult.job] = err
 		}
 	}
 
-	// TODO clean up job completely??
-
 	return cancelledJobs
 }
 
-func (repo *RedisJobRepository) Remove(jobIds []string) (cleanedJobIds []string, e error) {
+func processDeletionResponse(deletionResponse *deleteJobRedisResponse) (int64, error) {
+	var totalUpdates int64 = 0
+	var errorMessage error = nil
 
-	jobs, e := repo.getJobIdentities(jobIds)
+	modified, e := deletionResponse.removeFromLeasedResult.Result()
+	totalUpdates += modified
 	if e != nil {
-		return nil, e
+		errorMessage = e
```

Comment:
It might be worth to wrap the error here with description of what actually failed.